hass_nabucasa/ice_servers.py

```diff
@@ -80,7 +83,16 @@ async def _async_refresh_ice_servers(self) -> None:
                 if self._ice_servers_listener is not None:
                     await self._ice_servers_listener()
             except ClientResponseError as err:
-                _LOGGER.error("Can't refresh ICE servers: %s", err)
+                _LOGGER.error("Can't refresh ICE servers: %s", err.message)
+
+                # We should not keep the existing ICE servers with old timestamps
+                # as that will retrigger a refresh almost immediately.
+                if err.status in (401, 403):
+                    self._ice_servers = []
+
+                # we should update the listener as the ICE servers might have changed
+                if self._ice_servers_listener is not None:
```

Comment:
This is duplicated from L83-84
Do this is `finally`, or after the try/except instead.

Reply:
Good catch. Moved to `finally`.

Reply:
Good catch. 👍🏻 Moved it to `finally` first but realized after try/except is even better, so I moved it there.